File: LocationVoiture/src/main/java/model/Compte.java
package model;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name="type_compte",columnDefinition = "ENUM('Admin','Loueur','Client')")
@Table(name="compte")
public abstract class Compte {
	
	@JsonView(JsonViews.Common.class)
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	protected Integer id;
	
	@JsonView(JsonViews.Common.class)
	@Column(length = 100,nullable = false)
	protected String password;
	
	@JsonView(JsonViews.Common.class)
	@Column(length = 35,nullable = false,unique=true)
	protected String login;
	
	@JsonView(JsonViews.Common.class)
	@Column(length = 35,nullable = false)
	protected String nom;
	
	@JsonView(JsonViews.Common.class)
	@Column(length = 35,nullable = false)
	protected String prenom;
	
	@JsonView(JsonViews.Common.class)
	@Column(length = 35)
	protected Adresse adresse;
	
	public Compte() {
	}
	
	
	
	public Compte(String password, String login, String nom, String prenom) {
		super();
		this.password = password;
		this.login = login;
		this.nom = nom;
		this.prenom = prenom;
	}
	
	
	
	public Compte(String password, String login, String nom, String prenom, Adresse adresse) {
		super();
		this.password = password;
		this.login = login;
		this.nom = nom;
		this.prenom = prenom;
		this.adresse = adresse;
	}



	public String getPassword() {
		return password;
	}
	public void setPassword(String password) {
		this.password = password;
	}
	public String getLogin() {
		return login;
	}
	public void setLogin(String login) {
		this.login = login;
	}
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public String getNom() {
		return nom;
	}
	public void setNom(String nom) {
		this.nom = nom;
	}
	public String getPrenom() {
		return prenom;
	}
	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public Adresse getAdresse() {
		return adresse;
	}

	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}
	
	


}

File: LocationVoitureAngular/src/app/location/model/compte.ts
import { Adresse } from './adresse';

export class Compte {
  public get email(): string | undefined {
    return this._email;
  }
  public set email(value: string | undefined) {
    this._email = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  public get adresse(): Adresse | undefined {
    if (!this._adresse) {
      this._adresse = new Adresse();
    }
    return this._adresse;
  }
  public set adresse(value: Adresse | undefined) {
    this._adresse = value;
  }
  public get prenom(): string | undefined {
    return this._prenom;
  }
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }
  public get nom(): string | undefined {
    return this._nom;
  }
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  constructor(
    private _id?: number,
    private _email?: string,
    private _nom?: string,
    private _prenom?: string,
    private _adresse?: Adresse
  ) {}
}
